### DISCUSSION TITLE
Need help with hashmap via signal

### DISCUSSION THREAD

**Discussion by katopz** [DioxusLabs/dioxus#2336] [Q&A] [answered]
I stuck here for 2 days so it's about time to ask before giving up

refer to
https://github.com/gist-rs/katana/blob/2d29125d3425fe8d0b086461d7b84f486450f265/web/src/main.rs#L140-L142

Everything work fine but if I uncomment those 3 lines.

```rust
                                                // onclick: move |_| {
                                                //     kana_focus_state.set(kana.romaji.clone());
                                                // },
```
I will get an error
```
`kana_hashmap` does not live long enough
```

<img width="838" alt="image" src="https://github.com/DioxusLabs/dioxus/assets/97060/c183aa3f-3744-46dc-88f9-1c72e20e5996">

from my understand `kana_hashmap` will drop and won't work after `onclick` in next state so compiler yelling at me.
I try create child component, pass signals as props, etc but nothing work.
I assume that `HashMap` and the pattern I used are the bad guy here but before I give up I think I have to ask for help first.

Any hint is welcome, nothing serious tho; I just try to have fun coding there (but stuck 😅).
Thanks in advance

**Comment by ealmloff** [comment, marked as answer]
You need to clone `kana.romaji` before moving it into the closure. or clone maybe_kana:
```rust
onclick: {
    // Clone
    let romaji = kana.romaji.clone();
    // Then move the clone into the closure
    move || {kana_focus_state.set(romaji);}
}
```

**Reply by katopz** [reply]
Ohhh after fighting with last boss 
> move occurs because `romaji` has type `std::string::String`, which does not implement the `Copy` trait`.

Now I'm making working! Yeahh! So fun 🤩
```rust
onclick: {
    let romaji = kana.romaji.clone();
    move |_| {kana_focus_state.set(romaji.to_string());}
},
```

Many thanks! Will blog soon.
